### PR TITLE
tweak(datatrakWeb): RN-1438: Upgrade React Query to V4

### PR DESCRIPTION
**Pull request by hrazasalman**
### Changes:

- Upgrading to react-query to most latest compatible version.

### CODE REVIEW

File: packages/datatrak-web/src/__tests__/features/Reports/Reports.test.tsx [hunk 1]
```diff
@@ -244,10 +244,10 @@ describe('Reports', () => {
     );
     renderComponent(<Reports />);
     await userEvent.click(await screen.findByPlaceholderText('Select survey...'));
-    await userEvent.click(screen.getByText('Basic clinic data - Demo Land'));
+    await userEvent.click(await screen.findByText('Basic clinic data - Demo Land'));
```
Comment (alexd-bes) [inline packages/datatrak-web/src/__tests__/features/Reports/Reports.test.tsx:247]: This looks fine. I just wanted to check - for the user everything works as expected?

File: packages/datatrak-web/src/api/mutations/useLogin.ts [hunk 2]
```diff
@@ -35,15 +35,15 @@ export const useLogin = () => {
       },
       onSuccess: async ({ user }) => {
         await queryClient.invalidateQueries();
+        await queryClient.removeQueries();
+
         if (from) {
-          navigate(from, {
-            state: null,
-          });
+          console.log('Navigating to from:', from);
```
Comment (alexd-bes) [inline packages/datatrak-web/src/api/mutations/useLogin.ts:41]:
```suggestion
```

File: packages/datatrak-web/src/api/mutations/useLogin.ts [hunk 3]
```diff
@@ -35,15 +35,15 @@ export const useLogin = () => {
...
+          navigate(from, { state: null });
         } else {
           const path = user.projectId ? ROUTES.HOME : ROUTES.PROJECT_SELECT;
-          navigate(path, {
-            state: from,
-          });
+          console.log('Navigating to path:', path);
```
Comment (alexd-bes) [inline packages/datatrak-web/src/api/mutations/useLogin.ts:45]:
```suggestion
```

File: packages/datatrak-web/src/api/mutations/useLogin.ts [hunk 4]
```diff
@@ -35,15 +35,15 @@ export const useLogin = () => {
       },
       onSuccess: async ({ user }) => {
         await queryClient.invalidateQueries();
+        await queryClient.removeQueries();
```
Comment (alexd-bes) [inline packages/datatrak-web/src/api/mutations/useLogin.ts:38]: Is there a reason for this change?

File: packages/datatrak-web/src/api/mutations/useLogin.ts (hunk 2 above)
Comment (hrazasalman) [reply packages/datatrak-web/src/api/mutations/useLogin.ts]: Oops, I overlooked it 😅, have removed them now.

File: packages/datatrak-web/src/__tests__/features/Reports/Reports.test.tsx (hunk 1 above)
Comment (hrazasalman) [reply packages/datatrak-web/src/__tests__/features/Reports/Reports.test.tsx:247]: Yes, I have tested navigating the system and submitted a few responses, but I am a little nervous about it. Will we be doing regression testing after the merge?

**Pull request merged**